schemas/bigip-virtual-server_v0.1.8.json

```diff
@@ -0,0 +1,203 @@
+{
+  "$schema": "http://json-schema/org/schema#",
+  "id": "f5schemadb://bigip-virtual-server_v0.1.8.json",
+
+  "type": "object",
+
+  "definitions": {
+    "backendType": {
+      "type": "object",
+      "properties": {
+        "healthMonitors": {
+          "type": "array",
+          "items": { "$ref": "#/definitions/healthMonitorType" }
+        },
+        "serviceName": { "type": "string", "minLength": 1 },
+        "servicePort": { "$ref": "#/definitions/portType" }
+      },
+      "additionalProperties": false,
+      "required": [ "serviceName", "servicePort" ]
+    },
+    "frontendIAppType": {
+      "type": "object",
+      "properties": {
+        "iapp": { "type": "string", "minLength": 1 },
+        "iappOptions": {
+          "type": "object",
+          "patternProperties": {
+            "^[a-zA-Z0-9_-]+$": { "type": "string", "minLength": 1 }
+          },
+          "additionalProperties": false
+        },
+        "iappPoolMemberTable": {
+          "type": "object",
+          "properties": {
+            "name": { "type": "string", "minLength": 1 },
+            "columns": {
+              "type": "array",
+              "items": {
+                "oneOf": [
+                  { "$ref": "#/definitions/iappAddressType" },
+                  { "$ref": "#/definitions/iappPortType" },
+                  { "$ref": "#/definitions/iappValueType" }
+                ]
+              }
+            }
+          },
+          "additionalProperties": false,
+          "required": [ "name", "columns" ]
+        },
+        "iappTables": {
+          "type": "object",
+          "patternProperties": {
+            "^[a-zA-Z0-9_-]+$": { "$ref": "#/definitions/iappTableType" }
+          },
+          "additionalProperties": false
+        },
+        "iappVariables": {
+          "type": "object",
+          "patternProperties": {
+            "^[a-zA-Z0-9_-]+$": { "type": "string", "minLength": 1 }
+          },
+          "additionalProperties": false
+        },
+        "partition": { "type": "string", "minLength": 1 }
+      },
+      "additionalProperties": false,
+      "required": [ "partition", "iapp", "iappOptions", "iappVariables",
+                    "iappPoolMemberTable" ]
+    },
+    "frontendVSType": {
+      "type": "object",
+      "properties": {
+        "balance": { "type": "string", "minLength": 1},
+        "partition": { "type": "string", "minLength": 1 },
+        "mode": { "type": "string", "enum": [ "http", "tcp", "udp" ] },
+        "sslProfile": { "$ref": "#/definitions/sslProfileType" },
+        "virtualAddress": { "$ref": "#/definitions/virtualAddressType" },
+        "sourceAddressTranslation": {
+          "type": "object",
+          "properties": {
+            "type": { "type": "string", "enum": [ "automap", "none", "snat" ] },
+            "pool": { "type": "string", "minLength": 1}
```

Comment:
Just checking to make sure minLength=1 is correct (for automap/none settings).

Reply:
This field should be left out when configuring automap/none that's why only types is required. So if you do add pool to those types that is technically not correct but we will ignore it and create the vs anyway and log about it.